frontend/src/components/TemplateEditor/TemplateEditor.js

```diff
@@ -1200,8 +1200,9 @@ export default class TemplateEditor extends React.Component {
                     const {
                         kind,
                         metadata: { name, namespace },
+                        data,
                     } = resource
-                    if (kind === 'Secret') {
+                    if (kind === 'Secret' && (!data || !data['install-config.yaml'])) {
```

Comment:
I think the logic would be easier to read this way:
```suggestion
                    if (kind === 'Secret' && !(data && data['install-config.yaml'])) {
```
Or if something like this works in this context:
```suggestion
                    if (kind === 'Secret' && !data?.['install-config.yaml']) {
```